rust/libnewsboat/src/logger.rs

```diff
@@ -217,25 +217,25 @@ impl Logger {
 
         let mut files = self.files.lock().expect("Someone poisoned logger's mutex");
 
-        if level as isize <= self.get_loglevel() {
-            if let Some(ref mut logfile) = files.logfile {
-                let level = format!("{level}: ");
-
-                // Ignoring the error since checking every log() call will be too bothersome.
-                let _ = logfile.write_all(timestamp.as_bytes());
-                let _ = logfile.write_all(level.as_bytes());
-                let _ = logfile.write_all(data);
-                let _ = logfile.write_all(b"\n");
-            }
+        if level as isize <= self.get_loglevel()
+            && let Some(ref mut logfile) = files.logfile
```

Comment:
Looks like this use of a if-let-expression is only supported starting from Rust 1.88.0.
We might need to mark the old version of this line with `#[allow(clippy::collapsible_if)]` for a while (until our minimum supported Rust version is 1.88.0 or higher)

Reply:
Yup, I think suppressing the lint is our only option actually, so let's do that.